### ISSUE TITLE
click on a dropdown created with field and error 500

### ISSUE THREAD

**Issue by satyan01** [pluginsGLPI/fields#111]
http://itsm/glpi/plugins/fields/front/commondropdown.form.php?ddtype=standardchangecategoryfielddropdown&id=14

IIS warning
52.  -MODULE_SET_RESPONSE_ERROR_STATUS 

ModuleName
FastCgiModule 

Notification
EXECUTE_REQUEST_HANDLER 

HttpStatus
500 

HttpReason
Internal Server Error 

HttpSubStatus
0 

ErrorCode
The operation completed successfully.
 (0x0) 

ConfigExceptionInfo

**Comment by trasher** [comment]
Please give the output from log files

**Comment by satyan01** [comment]
Which log file please ?

**Comment by trasher** [comment]
`glpi/files/_log/php-errors.log`, `glpi/files/_log/sql-errors.log` and maybe PHP system logs.

**Comment by satyan01** [comment]
nothing relevant in log file

**Comment by satyan01** [comment]
except that 2016-12-14 10:53:48 [854@UNGSRVGVA118]
  From getEntitiesRestrictRequest() in C:\inetpub\wwwroot\glpi\inc\db.function.php line 1686
  getEntitiesRestrictRequest called without any `table` specified!

**Comment by trasher** [comment]
If there is an 500 error, there must be something in the logs. This is probably in you system files logs then.

**Comment by satyan01** [comment]
52.  -MODULE_SET_RESPONSE_ERROR_STATUS 


ModuleName
FastCgiModule 

Notification
EXECUTE_REQUEST_HANDLER 

HttpStatus
500 

HttpReason
Internal Server Error 

HttpSubStatus
0 

ErrorCode
The operation completed successfully.
 (0x0) 

ConfigExceptionInfo

**Comment by satyan01** [comment]
is it normal to have many cgi/fast cgi processes ?

**Comment by anachron13** [comment]
Hello,

Same problem here using GLPI 9.1.1 and Fields 1.4.1.
I got the following error in logs\php_error.log : 
`[19-Dec-2016 09:58:23 UTC] PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Attempt to load unsecure or missing C:\wamp\www\glpi/files/_plugins/fields/front/servicefielddropdown.form.php!' in C:\wamp\www\glpi\plugins\fields\front\commondropdown.form.php:8
Stack trace:
#0 {main}
  thrown in C:\wamp\www\glpi\plugins\fields\front\commondropdown.form.php on line 8`

**Comment by trasher** [comment]
Do you have files in `glpi/files/_plugins/fields/front`? What about `glpi/files/_plugins/fields/inc`?

**Comment by satyan01** [comment]
yes many files

![image](https://cloud.githubusercontent.com/assets/15144026/21310252/cfe2a176-c5e1-11e6-8e20-313eaf85c32b.png)

![image](https://cloud.githubusercontent.com/assets/15144026/21310258/dbfcf858-c5e1-11e6-91ce-4d022d1c4b35.png)

**Comment by anachron13** [comment]
The front folder contain `servicefielddropdown.form.php` and `servicefielddropdown.php` and the inc folder `servicefielddropdown.class.php`, `ticketservice.class.php`, `ticketserviceinjection.class.php`.

**Comment by trasher** [comment]
@satyan01 you have many files that must not be present there (`container.class.php`, `field.class.php` and so on). Those versions will sometimes override plugin ones, no way to predict what will happen. You'll have to clean up the dir and regenerate plugin files.

@anachron13 what is the content of `servicefielddropdown.form.php`? If the file is present, and its content OK; I guess it should be a windows specific issue... But I do not have windows.

**Comment by anachron13** [comment]
The content of the servicefielddropdown.form.php file : 
`<?php

$dropdown = new PluginFieldsServicefieldDropdown();
$options = [
   'ddtype' => $_GET['ddtype']
];
include GLPI_ROOT . "/front/dropdown.common.form.php";`

**Comment by satyan01** [comment]
still error 500 when I want to go to dropdown

**Comment by trasher** [comment]
@anachron13 sounds like this file is correct

I don't know why the file cannot be found. I suspect this is a windows issue; since both of you are using windows and I'm not able to reproduce.
As I do not use windows, I can't work on this.

@satyan01 the exact error is needed.

**Comment by satyan01** [comment]
![image](https://cloud.githubusercontent.com/assets/15144026/21313910/9c621c1a-c5f4-11e6-8fd2-3fb791c1e483.png)

**Comment by anachron13** [comment]
If it could help you resolve this issue, it work if i comment out the control on line 5. I don't really know anything in php so i don't know what is the purpose of this.

<?php

include "../../../inc/includes.php";
$path = PLUGINFIELDS_FRONT_PATH . '/' . $_REQUEST['ddtype'] . '.form.php';
//if (strpos(realpath($path), PLUGINFIELDS_FRONT_PATH) === 0) {
    include_once $path;
//} else {
 //   throw new \RuntimeException('Attempt to load unsecure or missing ' . $path .'!');
//}

**Comment by trasher** [comment]
Oh, I guess this is a `/` vs `\` issue then... Could you please add the following lines just before the if and give me the output you'll see on the page:
```
var_dump(realpath($path));
var_dump(PLUGINFIELDS_FRONT_PATH);
```

**Comment by anachron13** [comment]
It indeed look like it. Here is the result : 
```
C:\wamp\www\glpi\plugins\fields\front\commondropdown.form.php:5:string 'C:\wamp\www\glpi\files\_plugins\fields\front\servicefielddropdown.form.php' (length=74)
C:\wamp\www\glpi\plugins\fields\front\commondropdown.form.php:6:string 'C:\wamp\www\glpi/files/_plugins/fields/front' (length=44)
```

**Comment by trasher** [comment]
OK; I'll fix it asap; thanks for your tests :)

**Comment by satyan01** [comment]
content of the file

<?php

$dropdown = new PluginFieldsStandardchangecategoryfieldDropdown();
include GLPI_ROOT . "/front/dropdown.common.form.php";

**Comment by satyan01** [comment]
<?php

$dropdown = new PluginFieldsStandardchangecategoryfieldDropdown();
include GLPI_ROOT . "/front/dropdown.common.form.php";

**Comment by trasher** [comment]
See fix in e744f7c7dc2dff0c4e610c5675619ac0675af240

**Comment by satyan01** [comment]
thx it works
